internal/controller/ocirepository_controller.go

```diff
@@ -390,7 +391,8 @@ func (r *OCIRepositoryReconciler) reconcileSource(ctx context.Context, sp *patch
 	}
 
 	// Get the upstream revision from the artifact digest
-	revision, err := r.getRevision(url, opts.craneOpts)
+	// TODO: getRevision resolves the digest, which may change before image is fetched, so it should probaly update ref
```

Comment:
I suppose we should move this into an issue? I'm not inclined to address it in this PR, as it's a bug that was here before.

Reply:
https://github.com/fluxcd/source-controller/issues/1245